File: tests/unitTestRunner.server.lua
local DEVELOPER_ID = 1
local DEVELOPER_GAME_KEY = "cc5055f0-29e1-42d4-923f-11a00760a1a8"

local function runUnitModule(unitModule)
	local success, message = unitModule.run(DEVELOPER_ID, DEVELOPER_GAME_KEY)

	if not success then
		return warn(string.format("[UnitTest %s]: Fail (%s)", unitModule.Name, message or "<Unknown error>"))
	end
end

for _, unitTestModule in script.Parent.unitTests:GetChildren() do
	local unitModule = require(unitTestModule)

	task.spawn(runUnitModule, unitModule)
end

File: src/tests/ts/unitTestRunner.server.ts
export {};

const DEVELOPER_ID: number = 1
const DEVELOPER_GAME_KEY: string = "cc5055f0-29e1-42d4-923f-11a00760a1a8"

function runUnitModule(unitModule: { run(id: number, key: string): any, Name: string }) {
    let success, message = unitModule.run(DEVELOPER_ID, DEVELOPER_GAME_KEY)

    if (!success) {
        return warn(`[UnitTest ${unitModule.Name}]: Fail (${message})`)
    }
}


//@ts-ignore
for (let _ of script.Parent.GetChildren()) {
    task.spawn(() => {
        //@ts-ignore
        const unitModule: { run(id: number, key: string), Name: string } = require(`${_[2]}`)
        runUnitModule(unitModule)
    })
}

